src/compute/src/server.rs

```diff
@@ -464,8 +442,15 @@ pub async fn compute_node_serve(
 
     // All set, let the meta service know we're ready.
     meta_client.activate(&advertise_addr).await.unwrap();
+    // Wait for the shutdown signal.
+    let _ = shutdown.cancelled().await;
+
+    // TODO(shutdown): gracefully unregister from the meta service.
```

Comment:
I'm not sure if it's necessary to add this logic at the current stage, as it requires high reliability for auto scaling. But it's very elegant, looking forward to a time when it can be done fearlessly.